### PR TITLE
Add support for guaranteed memory field

### PR DESCRIPTION
**Pull request by eslutsky**

### CODE REVIEW

File: pkg/clients/ovirt/impl.go [hunk 1]
```diff
@@ -69,6 +69,11 @@ func (is *ovirtClient) CreateVMByMachine(
 		if providerSpec.MemoryMB > 0 {
 			vmBuilder.Memory(int64(math.Pow(2, 20)) * int64(providerSpec.MemoryMB))
```
Comment (Gal-Zaidman) [inline pkg/clients/ovirt/impl.go:70]: Please change `int64(math.Pow(2, 20)` to a constant like in the TF provider PR

**Comment by Gal-Zaidman** [comment]
We also need to add some validations.
1. GuaranteedMemoryMB must be set with MemoryMB.
2. GuaranteedMemoryMB can't be bigger than MemoryMB.

**Comment by eslutsky** [comment]
> 2\. GuaranteedMemoryMB


> We also need to add some validations.
> 
> 1. GuaranteedMemoryMB must be set with MemoryMB.
> 2. GuaranteedMemoryMB can't be bigger than MemoryMB
.
1. MemoryMB is mandatory and already validated [here](https://github.com/openshift/cluster-api-provider-ovirt/blob/179e13ad8479dfd65416e8a05bf72aa798732b05/pkg/actuators/machine/validations.go#L65):
2. done

File: pkg/clients/ovirt/impl.go (hunk 1 above)
Comment (eslutsky) [reply pkg/clients/ovirt/impl.go]: done

**Comment by Gal-Zaidman** [comment]
> > 2. GuaranteedMemoryMB
> 
> > We also need to add some validations.
> > 
> > 1. GuaranteedMemoryMB must be set with MemoryMB.
> > 2. GuaranteedMemoryMB can't be bigger than MemoryMB
> >    .
> 
> 1. MemoryMB is mandatory and already validated [here](https://github.com/openshift/cluster-api-provider-ovirt/blob/179e13ad8479dfd65416e8a05bf72aa798732b05/pkg/actuators/machine/validations.go#L65)
> 2. done

OK sure we want to remove InstanceTypeId so I'm good with that

File: pkg/actuators/machine/validations.go [hunk 2]
```diff
@@ -176,3 +181,17 @@ func versionCompare(v *ovirtsdk.Version, other *ovirtsdk.Version) (int64, error)
 	}
 	return result, nil
 }
+
+// validateGuaranteedMemory execute validation regarding the Virtual Machine validateGuaranteedMemory
+// Returns: nil or error
+func validateGuaranteedMemory(config *ovirtconfigv1.OvirtMachineProviderSpec) error {
+
```
Comment (Gal-Zaidman) [inline pkg/actuators/machine/validations.go:188]: extra new line

Comment (eslutsky) [reply pkg/actuators/machine/validations.go]: done

**Comment by eslutsky** [comment]
> > > 1. GuaranteedMemoryMB
> > 
> > 
> > > We also need to add some validations.
> > > 
> > > 1. GuaranteedMemoryMB must be set with MemoryMB.
> > > 2. GuaranteedMemoryMB can't be bigger than MemoryMB
> > >    .
> > 
> > 
> > 
> > 1. MemoryMB is mandatory and already validated [here](https://github.com/openshift/cluster-api-provider-ovirt/blob/179e13ad8479dfd65416e8a05bf72aa798732b05/pkg/actuators/machine/validations.go#L65)
> > 2. done
> 
> OK sure we want to remove InstanceTypeId so I'm good with that

done

File: pkg/clients/ovirt/impl.go [hunk 3]
```diff
@@ -67,7 +71,16 @@ func (is *ovirtClient) CreateVMByMachine(
...
 		if providerSpec.MemoryMB > 0 {
-			vmBuilder.Memory(int64(math.Pow(2, 20)) * int64(providerSpec.MemoryMB))
+			vmBuilder.Memory(int64(BytesInMB) * int64(providerSpec.MemoryMB))
+		}
+		if providerSpec.GuaranteedMemoryMB > 0 {
+			memoryPolicyBuilder := ovirtsdk.NewMemoryPolicyBuilder()
+			memoryPolicyBuilder.Guaranteed(int64(BytesInMB) * int64(providerSpec.GuaranteedMemoryMB))
+			memoryPolicy, err := memoryPolicyBuilder.Build()
+			if err != nil {
+				return nil, err
```
Comment (Gal-Zaidman) [inline pkg/clients/ovirt/impl.go:81]: Please wrap the error and provider context to the failure

Comment (eslutsky) [reply pkg/clients/ovirt/impl.go]: done

**Comment by eslutsky** [comment]
verified by adding the new field to the machine-set and performed scale-up causing new workers created with the new field.

**Pull request merged**